job/jobcache.go

```diff
@@ -42,9 +92,29 @@ func (c *JobCache) Set(j *Job) {
 	return
 }
 
-func (c *JobCache) Delete(id string) {
+func (c *MemoryJobCache) Delete(id string) {
 	c.rwLock.Lock()
 	defer c.rwLock.Unlock()
 
 	delete(c.Jobs, id)
 }
+
+func (c *MemoryJobCache) Persist() error {
+	c.rwLock.Lock()
+	defer c.rwLock.Unlock()
+
+	for _, v := range c.Jobs {
+		err := v.Save(c.jobDB)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *MemoryJobCache) PersistEvery(waitTime time.Duration) {
+	for {
+		time.Sleep(waitTime)
+		go c.Persist()
```

Comment:
@ajvb I'd instead do the following so that (1) calls to `c.Persist()` can't stack up (even though only one can run at a time; one could hang for a while, then there could be a bunch of quick ones in rapid succession, which you probably don't want), (2) so that this actually runs every `waitTime` moments, not `waitTime` + however long it takes for `c.Persist()` to execute, and (3) so that you handle the error:

```go
wait := time.Tick(waitTime)
var err error
for {
    <-wait
    err = c.Persist()
    if err != nil {
        // Handle error
    }
}
```